src/Std/Time.lean

```diff
@@ -0,0 +1,174 @@
+represents a day in a month, ranging from 1 to 31. Some ordinal types like `Hour.Ordinal` and `Second.Ordinal`,
+allow for values beyond the normal range (e.g, 24 hours and 61 seconds) to accomodate special cases
+with leap seconds like `24:00:00` that is valid in ISO 8601.
+
+- Ordinal types:
+  - `Day.Ordinal`: Ranges from 1 to 31.
+  - `Day.Ordinal.OfYear`: Ranges from 1 to (365 or 366).
+  - `Month.Ordinal`: Ranges from 1 to 12.
+  - `WeekOfYear.Ordinal`: Ranges from 1 to 53.
+  - `Hour.Ordinal`: Ranges from 0 to 24.
+  - `Millisecond.Ordinal`: Ranges from 0 to 999.
+  - `Minute.Ordinal`: Ranges from 0 to 59.
+  - `Nanosecond.Ordinal`: Ranges from 0 to 999,999,999.
+  - `Second.Ordinal`: Ranges from 0 to 60.
+  - `Weekday`: That is a inductive type with all the seven days.
+
+## Span
+
+Span types are used as subcomponents of other types. They represent a range of values in the limits
+of the parent type, e.g, `Nanosecond.Span` ranges from -999,999,999 to 999,999,999, as 1,000,000,000
+nanoseconds corresponds to one second.
+
+- Span types:
+  - `Nanosecond.Span`: Ranges from -999,999,999 to 999,999,999.
+
+# Date and Time Types
+
+Dates and times are composed of these components. Dates are "absolute" value in contrast with Offsets
+that are just shifts in dates and times. Types like `Date` are made using of components such as `Year.Offset`,
+`Month.Ordinal`, and `Day.Ordinal`, with a proof of the date's validity. Some types, like `Date.Scalar`,
+are wrappers around offsets (e.g., `Day.Offset`) and represent absolute dates relative to the UNIX Epoch.
+
+## Date
+These types provide precision down to the day level, useful for representing and manipulating dates.
+
+- **`LocalDate`:** Represents a calendar date in the format `YYYY-MM-DD`.
+- **`LocalDate.Scalar`:** : Represents an absolute date with day-level precision in a compact format `DD`.
+  It encapsulates days since the UNIX Epoch.
+- **`WeekDate`:** Uses the `YYYY-Www` format with week level precision.
+
+## Time
+These types offer precision down to the nanosecond level, useful for representing and manipulating time of day.
+
+- **`LocalTime`**: Represents a time of day in the format `HH:mm:ss.SSSSSSSSS`.
+
+## Date and time
+Combines date and time into a single representation, useful for precise timestamps and scheduling.
+
+- **`LocalDateTime`**: Represents both date and time in the format `YYYY-MM-DDTHH:mm:ss.SSSSSSSSS`.
+- **`Timestamp`**: Represents a point in time with second-level precision.
+
+## Zoned date and times.
+Combines date, time and time zones.
+
+- **`DateTime`**: Represents both date and time but with a time zone in the type constructor.
+- **`ZonedDateTime`**: An existencial version of the `DateTime`.
+
+## Duration
+Represents spans of time and the difference between two points in time.
+
+- **`Instant`**: Represents a non-negative instant in time with nanosecond precision, represents a
```

Comment:
So is the difference between `LocalDateTime` and `Instant` that the former refers to "the thing you see when you look at a (very high-precision) clock" while the other represents the abstract concept of time?